src/main/java/jimmy/mcgymmy/logic/predicate/DatePredicate.java

```diff
@@ -0,0 +1,44 @@
+package jimmy.mcgymmy.logic.predicate;
+
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+import jimmy.mcgymmy.commons.core.LogsCenter;
+import jimmy.mcgymmy.logic.parser.ParserUtil;
+import jimmy.mcgymmy.logic.parser.exceptions.ParseException;
+import jimmy.mcgymmy.model.date.Date;
+import jimmy.mcgymmy.model.food.Food;
+
+/**
+ * Tests that a {@code Food}'s {@code Date} matches the {@code String} date given.
+ */
+public class DatePredicate implements Predicate<Food> {
+    private final Date date;
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    /**
+     * Initialises {@code DatePredicate} object.
+     */
+    public DatePredicate(String date) throws ParseException {
+        Date date1 = null;
+        try {
+            Date parsedDate = ParserUtil.parseDate(date);
+            date1 = parsedDate;
```

Comment:
I think changing it to this will make the code shorter:
```
this.date = null;
try {
    date = ParserUtil.parseDate(date);
}
[rest of the code]
}

Reply:
Thanks, this was auto generated code I think, will look to refactor other bits as well soon.